### DISCUSSION TITLE
Extending Prisma generated TypeScript models?

### DISCUSSION THREAD

**Discussion by tarazena** [prisma/prisma#11858] [Q&A] [answered]
Hi

I have a Prisma schema that generates User model
```
model User {
  email                   String?
  firstName               String?
  id                      String               @id @default(uuid())
  lastName                String?
}
```
What I want to do is that I want to create a function that returns full name like
```ts
get fullName() =>  `${this.firstName} ${this.lastName}`
```

where `this` is the instance that is being returned from Prisma like
```
const user = prisma.user.findUnique({
  where: {
      id: '1234'
  }
});

const fullName = user.fullName();
```

Can this be done? I tried something like
```
import { User } from 'prisma';

type IUserClass = User;

class UserClass implements IUserClass {
  email?: string;
  firstName?: string;
  id: string;
  lastName?: string;
 
  constructor(user: User) {
    Object.assign(this, user);
  }

  get fullName = { return `${this.firstName} ${this.lastName}` }
}
```

While the code above works, it does not scale well, So I was wondering if there is a way to make it cleaner.

**Comment by austincrim** [comment, marked as answer]
Hey @tarazena 👋 ,

This is not yet possible, but we have an [outstanding feature request](https://github.com/prisma/prisma/issues/7161) that would cover this use case.

Feel free to give it a thumbs up or add your opinion on how you want it to look!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
